#### About

> Index numbers and Inflation

Index numbers are statistical measures that allow us to track changes in the value of a variable over time. 

Inflation is the rate at which the general level of prices for goods and services is rising, and subsequently, purchasing power is falling. 

Let's say we have data on the prices of different commodities over the years. We can calculate the price index for each year, which will give us an idea of the changes in prices over time. The formula for calculating the price index is:

Price index = (Price of commodity in current year / Price of commodity in base year) x 100

Let's assume 2020 as the base year and have the following data:

In [1]:
import pandas as pd

data = {'Year': [2018, 2019, 2020, 2021],
        'Price of Rice': [30, 35, 40, 50],
        'Price of Wheat': [25, 28, 30, 35]}
df = pd.DataFrame(data)
df.set_index('Year', inplace=True)

print(df)

      Price of Rice  Price of Wheat
Year                               
2018             30              25
2019             35              28
2020             40              30
2021             50              35


Now, we can calculate the price index for each year using the above formula:



In [2]:
base_year = 2020
df['Price Index Rice'] = (df['Price of Rice'] / df.loc[base_year, 'Price of Rice']) * 100
df['Price Index Wheat'] = (df['Price of Wheat'] / df.loc[base_year, 'Price of Wheat']) * 100

print(df)

      Price of Rice  Price of Wheat  Price Index Rice  Price Index Wheat
Year                                                                    
2018             30              25              75.0          83.333333
2019             35              28              87.5          93.333333
2020             40              30             100.0         100.000000
2021             50              35             125.0         116.666667


> Conclusion


From the above output, we can observe that the price index of rice and wheat increased every year. In 2018, the price index of rice was 75% and the price index of wheat was 83.33% of the base year. Similarly, in 2021, the price index of rice was 125% and the price index of wheat was 116.67% of the base year.



Calculating Inflation rate

In [3]:
df['Inflation Rate Rice'] = ((df['Price Index Rice'] - 100) / 100) * 100
df['Inflation Rate Wheat'] = ((df['Price Index Wheat'] - 100) / 100) * 100

print(df)

      Price of Rice  Price of Wheat  Price Index Rice  Price Index Wheat   
Year                                                                       
2018             30              25              75.0          83.333333  \
2019             35              28              87.5          93.333333   
2020             40              30             100.0         100.000000   
2021             50              35             125.0         116.666667   

      Inflation Rate Rice  Inflation Rate Wheat  
Year                                             
2018                -25.0            -16.666667  
2019                -12.5             -6.666667  
2020                  0.0              0.000000  
2021                 25.0             16.666667  



The price index for rice in 2018 is 75.0, which means that the price of rice in 2018 was 75% of the price of rice in the base year. Similarly, the price index for wheat in 2018 is 83.333333, which means that the price of wheat in 2018 was 83.33% of the price of wheat in the base year.